examples/image.rs

```diff
@@ -13,14 +13,22 @@ use std::{boxed, env, error, fs, io};
 mod common;
 
 fn main() {
-    let dkr_ref = match std::env::args().nth(1) {
-        Some(ref x) => reference::Reference::from_str(x),
-        None => reference::Reference::from_str("quay.io/coreos/etcd"),
-    }
-    .unwrap();
-    let registry = dkr_ref.registry();
+    let registry = match std::env::args().nth(1) {
+        Some(x) => x,
+        None => "quay.io".into(),
+    };
 
-    println!("[{}] downloading image {}", registry, dkr_ref);
+    let image = match std::env::args().nth(2) {
+        Some(x) => x,
+        None => "coreos/etcd".into(),
+    };
+
+    let version = match std::env::args().nth(3) {
```

Comment:
I'm not sure I'm getting the reason behind splitting registry / image / version here. Is reference parsing broken on some input?

Reply:
It seems so: https://github.com/camallo/dkregistry-rs/pull/93

Reply:
Ack. I started massaging the parsing logic there to take care of such case, but let's keep this split here in the meanwhile.